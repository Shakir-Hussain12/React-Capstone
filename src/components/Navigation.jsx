import React from 'react';
import {
  Container, Form, Nav, Navbar,
} from 'react-bootstrap';
import PropTypes from 'prop-types';
import { NavLink } from 'react-router-dom';
import { MdOutlineArrowBackIosNew } from 'react-icons/md';
import { BsMic } from 'react-icons/bs';
import { FiSettings } from 'react-icons/fi';

function Navigation({ url, back }) {
  return (
    <Navbar style={{ background: '#1f5ca3' }}>
      <Container fluid>
        <Navbar.Brand style={{ marginInlineStart: '2%', marginTop: '-2%' }}>
          <NavLink to={back}>
            <MdOutlineArrowBackIosNew style={{ color: 'white' }} />
          </NavLink>
        </Navbar.Brand>
        <Nav
          style={{ maxHeight: '100px' }}
          navbarScroll
        >
          <Form className="d-flex">
            <Form.Control
              type="search"
              placeholder={`${url}/`}
              className="me-5"
              aria-label="Search"
            />
          </Form>
          <Nav.Link href="/">
            <BsMic style={{ color: 'white' }} />
          </Nav.Link>
          <Nav.Link href="/">
            <FiSettings style={{ color: 'white' }} />
          </Nav.Link>
        </Nav>
      </Container>
    </Navbar>
  );
}

export default Navigation;

Navigation.defaultProps = {
  url: '',
  back: '..',
};

Navigation.propTypes = {
  url: PropTypes.string,
  back: PropTypes.string,
};
