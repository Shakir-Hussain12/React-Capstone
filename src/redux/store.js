import { configureStore } from '@reduxjs/toolkit';
import quranReducer from './Quran/quranReducer';
import verseReducer from './Verse/verseReducer';

const store = configureStore({
  reducer: {
    Quran: quranReducer,
    Verse: verseReducer,
  },
});

export default store;
