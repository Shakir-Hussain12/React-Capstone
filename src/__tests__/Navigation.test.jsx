import React from 'react';
import { render } from '@testing-library/react';
import { Provider } from 'react-redux';
import { BrowserRouter } from 'react-router-dom';
import '@testing-library/jest-dom';
import store from '../redux/store';
import Navigation from '../components/Navigation';

describe('App', () => {
  it('renders Navigation', () => {
    const tree = render(
      <Provider store={store}>
        <BrowserRouter>
          <Navigation />
        </BrowserRouter>
      </Provider>,
    );

    expect(tree).toMatchSnapshot();
  });
});
