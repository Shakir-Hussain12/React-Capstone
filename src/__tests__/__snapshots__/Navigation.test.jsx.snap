// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`App > renders Navigation 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <nav
        class="navbar navbar-expand navbar-light bg-light"
      >
        <div
          class="container-fluid"
        >
          <span
            class="navbar-brand"
            style="margin-inline-start: 2%; margin-top: -2%;"
          >
            <a
              aria-current="page"
              class="active"
              href="/"
            >
              <svg
                fill="currentColor"
                height="1em"
                stroke="currentColor"
                stroke-width="0"
                viewBox="0 0 24 24"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M0 0h24v24H0z"
                  fill="none"
                />
                <path
                  d="M17.77 3.77L16 2 6 12l10 10 1.77-1.77L9.54 12z"
                />
              </svg>
            </a>
          </span>
          <div
            class="navbar-nav navbar-nav-scroll"
            style="max-height: 100px;"
          >
            <form
              class="d-flex"
            >
              <input
                aria-label="Search"
                class="me-5 form-control"
                placeholder="/"
                type="search"
              />
            </form>
            <a
              class="nav-link"
              data-rr-ui-event-key="/"
              href="/"
            >
              <svg
                fill="currentColor"
                height="1em"
                stroke="currentColor"
                stroke-width="0"
                viewBox="0 0 16 16"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M3.5 6.5A.5.5 0 0 1 4 7v1a4 4 0 0 0 8 0V7a.5.5 0 0 1 1 0v1a5 5 0 0 1-4.5 4.975V15h3a.5.5 0 0 1 0 1h-7a.5.5 0 0 1 0-1h3v-2.025A5 5 0 0 1 3 8V7a.5.5 0 0 1 .5-.5z"
                />
                <path
                  d="M10 8a2 2 0 1 1-4 0V3a2 2 0 1 1 4 0v5zM8 0a3 3 0 0 0-3 3v5a3 3 0 0 0 6 0V3a3 3 0 0 0-3-3z"
                />
              </svg>
            </a>
            <a
              class="nav-link"
              data-rr-ui-event-key="/"
              href="/"
            >
              <svg
                fill="none"
                height="1em"
                stroke="currentColor"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="2"
                viewBox="0 0 24 24"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <circle
                  cx="12"
                  cy="12"
                  r="3"
                />
                <path
                  d="M19.4 15a1.65 1.65 0 0 0 .33 1.82l.06.06a2 2 0 0 1 0 2.83 2 2 0 0 1-2.83 0l-.06-.06a1.65 1.65 0 0 0-1.82-.33 1.65 1.65 0 0 0-1 1.51V21a2 2 0 0 1-2 2 2 2 0 0 1-2-2v-.09A1.65 1.65 0 0 0 9 19.4a1.65 1.65 0 0 0-1.82.33l-.06.06a2 2 0 0 1-2.83 0 2 2 0 0 1 0-2.83l.06-.06a1.65 1.65 0 0 0 .33-1.82 1.65 1.65 0 0 0-1.51-1H3a2 2 0 0 1-2-2 2 2 0 0 1 2-2h.09A1.65 1.65 0 0 0 4.6 9a1.65 1.65 0 0 0-.33-1.82l-.06-.06a2 2 0 0 1 0-2.83 2 2 0 0 1 2.83 0l.06.06a1.65 1.65 0 0 0 1.82.33H9a1.65 1.65 0 0 0 1-1.51V3a2 2 0 0 1 2-2 2 2 0 0 1 2 2v.09a1.65 1.65 0 0 0 1 1.51 1.65 1.65 0 0 0 1.82-.33l.06-.06a2 2 0 0 1 2.83 0 2 2 0 0 1 0 2.83l-.06.06a1.65 1.65 0 0 0-.33 1.82V9a1.65 1.65 0 0 0 1.51 1H21a2 2 0 0 1 2 2 2 2 0 0 1-2 2h-.09a1.65 1.65 0 0 0-1.51 1z"
                />
              </svg>
            </a>
          </div>
        </div>
      </nav>
    </div>
  </body>,
  "container": <div>
    <nav
      class="navbar navbar-expand navbar-light bg-light"
    >
      <div
        class="container-fluid"
      >
        <span
          class="navbar-brand"
          style="margin-inline-start: 2%; margin-top: -2%;"
        >
          <a
            aria-current="page"
            class="active"
            href="/"
          >
            <svg
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              stroke-width="0"
              viewBox="0 0 24 24"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M0 0h24v24H0z"
                fill="none"
              />
              <path
                d="M17.77 3.77L16 2 6 12l10 10 1.77-1.77L9.54 12z"
              />
            </svg>
          </a>
        </span>
        <div
          class="navbar-nav navbar-nav-scroll"
          style="max-height: 100px;"
        >
          <form
            class="d-flex"
          >
            <input
              aria-label="Search"
              class="me-5 form-control"
              placeholder="/"
              type="search"
            />
          </form>
          <a
            class="nav-link"
            data-rr-ui-event-key="/"
            href="/"
          >
            <svg
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              stroke-width="0"
              viewBox="0 0 16 16"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M3.5 6.5A.5.5 0 0 1 4 7v1a4 4 0 0 0 8 0V7a.5.5 0 0 1 1 0v1a5 5 0 0 1-4.5 4.975V15h3a.5.5 0 0 1 0 1h-7a.5.5 0 0 1 0-1h3v-2.025A5 5 0 0 1 3 8V7a.5.5 0 0 1 .5-.5z"
              />
              <path
                d="M10 8a2 2 0 1 1-4 0V3a2 2 0 1 1 4 0v5zM8 0a3 3 0 0 0-3 3v5a3 3 0 0 0 6 0V3a3 3 0 0 0-3-3z"
              />
            </svg>
          </a>
          <a
            class="nav-link"
            data-rr-ui-event-key="/"
            href="/"
          >
            <svg
              fill="none"
              height="1em"
              stroke="currentColor"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="2"
              viewBox="0 0 24 24"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <circle
                cx="12"
                cy="12"
                r="3"
              />
              <path
                d="M19.4 15a1.65 1.65 0 0 0 .33 1.82l.06.06a2 2 0 0 1 0 2.83 2 2 0 0 1-2.83 0l-.06-.06a1.65 1.65 0 0 0-1.82-.33 1.65 1.65 0 0 0-1 1.51V21a2 2 0 0 1-2 2 2 2 0 0 1-2-2v-.09A1.65 1.65 0 0 0 9 19.4a1.65 1.65 0 0 0-1.82.33l-.06.06a2 2 0 0 1-2.83 0 2 2 0 0 1 0-2.83l.06-.06a1.65 1.65 0 0 0 .33-1.82 1.65 1.65 0 0 0-1.51-1H3a2 2 0 0 1-2-2 2 2 0 0 1 2-2h.09A1.65 1.65 0 0 0 4.6 9a1.65 1.65 0 0 0-.33-1.82l-.06-.06a2 2 0 0 1 0-2.83 2 2 0 0 1 2.83 0l.06.06a1.65 1.65 0 0 0 1.82.33H9a1.65 1.65 0 0 0 1-1.51V3a2 2 0 0 1 2-2 2 2 0 0 1 2 2v.09a1.65 1.65 0 0 0 1 1.51 1.65 1.65 0 0 0 1.82-.33l.06-.06a2 2 0 0 1 2.83 0 2 2 0 0 1 0 2.83l-.06.06a1.65 1.65 0 0 0-.33 1.82V9a1.65 1.65 0 0 0 1.51 1H21a2 2 0 0 1 2 2 2 2 0 0 1-2 2h-.09a1.65 1.65 0 0 0-1.51 1z"
              />
            </svg>
          </a>
        </div>
      </div>
    </nav>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
